ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java

```diff
@@ -267,6 +267,9 @@ private boolean getNextPath() throws Exception {
       if (isNonNativeTable) {
         return true;
       }
+      if (currPath.toUri().toString().contains("hadoop-dw2-nn.smf1.twitter.com")) {
```

Comment:
Too hacky to be merged.

Please extract the pattern from the path. Twdoop URI follow `hadoop-{cluster}-nn.{dc}.twitter.com` pattern. You can then transform this URI to {rootURI}/{dc}/{cluster}/path